src/process_util.rs

```diff
@@ -135,8 +153,11 @@ pub fn run_and_wait(cmd: &mut Command)
                         Stop(..) | Continue(..) | Death(..) => { }
                     }
                 }
-                try!(tty_guard.check().map_err(|e|
-                    format!("Error handling tty: {}", e)));
+                if take_tty {
+                    try!(tty_guard.check().map_err(|e|
+                        format!("Error handling tty: {}", e)));
+
+                }
             }
             _ => unreachable!(),
```

Comment:
Still crashes on SIGSTTIN, SIGTTOU, right?

I mean all trapped singnals should be handled somehow.